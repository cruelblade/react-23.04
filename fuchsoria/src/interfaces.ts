export interface IMessage {
  id: string;
  author?: string;
  text: string;
  authorAccess?: string;
  isSelf?: boolean;
  date?: number;
}

export interface IMessageWithHandler extends IMessage {
  handleDelete: () => void;
}

export interface IChatListItem {
  id: number | string;
  title: string;
  description: string;
}

export interface IChatListProps {
  items: IChatListItem[];
  blinkingIds?: string[];
}

export interface IChatContainerProps {
  chats: IChats;
  profile: IProfileState;
  addChat: (chatName: string, chatId: string) => void;
  addMessage: (message: IMessage, chatId: string, callback?: Function) => void;
  noContent?: boolean;
  history?: unknown;
  location?: unknown;
  match?: {
    isExact: boolean;
    params: {
      chatId: string;
    };
    path: string;
    url: string;
  };
}

export interface IProfileState {
  nickName: string;
}

export interface IChatContainerState {
  chatList?: IChatListItem[];
}

export interface IMessageListProps {
  messages: IMessage[];
  chatId: string;
}

export interface IChatFormProps {
  handleSubmit: (text: string, author?: string) => void;
}

export interface IChat {
  title: string;
  messages: IMessage[];
}

export interface IChats {
  [key: string]: IChat;
}

export interface IUpdateProfile {
  nickName?: string;
}

export interface IProfileFormProps {
  updateProfile: ({ nickName }: IUpdateProfile) => void;
}
