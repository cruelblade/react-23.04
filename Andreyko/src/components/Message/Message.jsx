import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './style.css'

export default class Message extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { text, author } = this.props;

    return (
      <div className={cx("message", {"my-message": author === 'me'})}>
        <div className="message-text">{text}</div>
        <div className="message-author">{author}</div>
      </div>
    )
  }
}

Message.defaultProps = {
  text: "Всегда существую!",
  author: "Создатель"
};

Message.propTypes = {
  text: PropTypes.string.isRequired,
  author: PropTypes.string,
};