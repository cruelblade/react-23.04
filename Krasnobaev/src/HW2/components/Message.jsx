import React, { Component } from "react";

class Message extends Component {
    state = {
        text: "",
        author: "Ivan K.",
    };

    onChange = (e) => {
        const { value, name } = e.target;
        this.setState({ [name]: value });
    };

    onSubmit = (e) => {
        const { addNewMessage } = this.props;
        e.preventDefault();
        addNewMessage(this.state.text, this.state.author);
        this.state.text = '';
    };

    render() {
        const { text, author } = this.state;

        return (
        <form onSubmit={this.onSubmit}>
            <label htmlFor="author">Author: </label>
            <input id="author" type="text" name="author" onChange={this.onChange} value={author} />
            <br/><br/>
            <label htmlFor="text">Your message: </label>
            <textarea id="text" name="text" onChange={this.onChange} value={text} />
            <button type="submit">Add message</button>
        </form>
        );
    }
}

export default Message;