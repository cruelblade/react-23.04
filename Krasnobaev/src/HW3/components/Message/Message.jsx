import React, { Component } from 'react';
import Button from '@material-ui/core/Button';
import Input from '@material-ui/core/Input';
import SendIcon from '@material-ui/icons/Send';
import styles from './index.css';

class Message extends Component {
    state = {
        text: "",
        author: "Ivan K.",
    };

    onChange = (e) => {
        const { value, name } = e.target;
        this.setState({ [name]: value });
    };

    semdMsg = () => {
        const { addNewMessage } = this.props;
        const { text, author } = this.state;

        
        addNewMessage({ text, author });
        this.state.text = '';
    };

    onSubmit = (e) => {
        e.preventDefault();
        this.semdMsg();
    };

    onKeyDown = (e) => {
        if (e.key === 'Enter' && (e.ctrlKey || e.metaKey)) {
            this.sendMsg();
        }
    }

    render() {
        const { text, author } = this.state;

        return (
        <form className={styles.container} noValidate onSubmit={this.onSubmit}>
            <Input 
                placeholder="Author:" 
                inputProps={{ 'aria-label': 'description' }} 
                name="author"
                onChange={this.onChange}
                value={author}
                className={styles.input}
            />
            <Input 
                placeholder="Your message:" 
                inputProps={{ 'aria-label': 'description' }} 
                name="text"
                onChange={this.onChange}
                onKeyDown={this.onKeyDown}
                value={text}
                className={styles.input}
            />
            <Button
                type="submit"
                variant="contained"
                color="primary"
                endIcon={<SendIcon />}
            >
                Add message 
            </Button>
        </form>
        );
    }
}

export default Message;