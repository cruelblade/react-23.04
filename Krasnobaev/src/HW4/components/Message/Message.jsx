import React, { Component } from 'react';
import Button from '@material-ui/core/Button';
import Input from '@material-ui/core/Input';
import SendIcon from '@material-ui/icons/Send';
import styles from './index.css';

class Message extends Component {
    state = {
        chats: {
            1: { title: "ChatRoom 1", messageList: [1] },
            2: { title: "ChatRoom 2", messageList: [1] },
        },
        messages: {
            1: { text: "Obey!", author: "Bot" },
        },
        newMsg: { text: "", author: "Ivan K."}
    };

    componentDidUpdate() {
        if (this.state.messages[messages.length - 1].author !== "Bot") {
            setTimeout(
                () => this.addNewMessage(this.state.messages[1]),
                500
            );
        }
    };
    
    addNewMessage = msg => {
        const { cid } = this.props;
        const newMsgId = Object.keys(this.state.messages).length + 1;

        this.setState(({ messages, chats }) => ({
        messages: {
            ...messages,
            [newMsgId]: { text: msg.text, author: msg.author },
        },
        chats: {
            ...chats,
            [cid]: {
            ...chats[cid],
            messageList: [...chats[cid].messageList, messageId],
            },
        },
        }));
    };

    onChange = (e) => {
        const { value, name } = e.target;
        
        this.setState( ({newMsg}) => ({
            newMsg: { text: value, author: name}
        }));
    };

    semdMsg = () => {
        const { text, author } = this.state.newMsg;
        
        this.addNewMessage({ text, author });
        this.state.newMsg.text = '';
    };

    onSubmit = (e) => {
        e.preventDefault();
        this.semdMsg();
    };

    onKeyDown = (e) => {
        if (e.key === 'Enter' && (e.ctrlKey || e.metaKey)) {
            this.sendMsg();
        }
    }

    render() {
        const { messages } = this.state;//передать все сообщения по id чата


        return (
            <div>
                <ChatList messages={messages} />
                <form className={styles.container} noValidate onSubmit={this.onSubmit}>
                    <Input
                        placeholder="Author:"
                        inputProps={{ 'aria-label': 'description' }}
                        name="author"
                        onChange={this.onChange}
                        value={author}
                        className={styles.input}
                    />
                    <Input
                        placeholder="Your message:"
                        inputProps={{ 'aria-label': 'description' }}
                        name="text"
                        onChange={this.onChange}
                        onKeyDown={this.onKeyDown}
                        value={text}
                        className={styles.input}
                    />
                    <Button
                        type="submit"
                        variant="contained"
                        color="primary"
                        endIcon={<SendIcon />}
                    >
                        Add message
                    </Button>
                </form>
            </div>
        );
    }
}

export default Message;