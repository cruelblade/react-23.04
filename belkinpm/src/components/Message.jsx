import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Message extends Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  };

  render() {
    return (
      <div
        className='message'
        style={{
          alignSelf: this.props.name === 'bot' ? 'flex-start' : 'flex-end',
          backgroundColor:
            this.props.name === 'bot' ? 'lightblue' : 'lightsteelblue',
          borderBottomLeftRadius: this.props.name === 'bot' ? '0' : '0.75rem',
          borderBottomRightRadius: this.props.name === 'bot' ? '0.75rem' : '0',
        }}
      >
        <div>{this.props.text}</div>
        <div className='message-sender'>{this.props.name}</div>
      </div>
    );
  }
}

export default Message;
