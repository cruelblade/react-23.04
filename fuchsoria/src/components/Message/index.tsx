import React from 'react';
import cn from 'classnames';
import { IMessageWithHandler } from '../../interfaces';
import styles from './styles.module.scss';

export default function Message({ author, text, isSelf, handleDelete }: IMessageWithHandler) {
  return (
    <div className={cn([styles.message, { [styles.messageSelf]: isSelf }])} onDoubleClick={handleDelete}>
      <div className={styles.messageAuthor}>{author}</div>
      {text}
    </div>
  );
}
