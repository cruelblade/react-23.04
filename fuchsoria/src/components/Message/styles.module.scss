.message {
  padding: 10px 20px;
  margin-bottom: 10px;
  text-align: left;
  align-self: flex-start;
  background-color: #ffff;
  border-radius: 10px;
  max-width: 400px;

  &:last-child {
    margin-bottom: 0;
  }

  &_self {
    text-align: right;
    align-self: flex-end;
  }

  &__author {
    user-select: none;
    text-transform: capitalize;
    font-weight: 700;
  }
}
