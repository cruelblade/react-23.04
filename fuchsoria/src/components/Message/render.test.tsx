import React from 'react';
import { render } from '@testing-library/react';
import { IMessage } from '../../interfaces';
import Message from '../Message';

const mockData: IMessage = {
  id: 'w7i33e48tr834',
  author: 'Jest',
  text: 'Jest example message',
  authorAccess: 'user',
};

const mockDataSelf: IMessage = {
  id: 'w7i3hgf48trg4',
  author: 'Self',
  text: 'Self example message',
  authorAccess: 'self',
};

const message = document.createElement('div');
const messageBot = document.createElement('div');

describe('<Message />', () => {
  it('Component should render', async () => {
    render(
      <Message
        id={mockData.id}
        key={mockData.id}
        author={mockData.author}
        text={mockData.text}
        isSelf={mockData.authorAccess === 'self'}
        handleDelete={jest.fn()}
      />,
      { container: document.body.appendChild(message) }
    );

    render(
      <Message
        id={mockData.id}
        key={mockDataSelf.id}
        author={mockDataSelf.author}
        text={mockDataSelf.text}
        isSelf={mockDataSelf.authorAccess === 'self'}
        handleDelete={jest.fn()}
      />,
      { container: document.body.appendChild(messageBot) }
    );

    expect(document.body).toMatchSnapshot();
  });

  it('Check rendered user message', () => {
    expect(message.querySelector('.messageAuthor')?.textContent).toBe(mockData.author);
    expect(message.querySelector('.message')?.lastChild?.textContent).toBe(mockData.text);
    expect(message.querySelector('.message')?.classList.contains('messageSelf')).toBeFalsy();
  });

  it('Check rendered self message', () => {
    expect(messageBot.querySelector('.messageAuthor')?.textContent).toBe(mockDataSelf.author);
    expect(messageBot.querySelector('.message')?.lastChild?.textContent).toBe(mockDataSelf.text);
    expect(messageBot.querySelector('.message')?.classList.contains('messageSelf')).toBeTruthy();
  });
});
