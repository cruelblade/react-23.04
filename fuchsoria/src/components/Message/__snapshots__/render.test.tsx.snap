// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> Component should render 1`] = `
<body>
  <div>
    <div
      class="message"
    >
      <div
        class="messageAuthor"
      >
        Jest
      </div>
      Jest example message
    </div>
  </div>
  <div>
    <div
      class="message messageSelf"
    >
      <div
        class="messageAuthor"
      >
        Self
      </div>
      Self example message
    </div>
  </div>
</body>
`;
